src/NCronJob/Scheduler/TaskScheduler/TaskFactoryProvider.cs

```diff
@@ -0,0 +1,13 @@
+namespace NCronJob;
+
+internal static class TaskFactoryProvider
+{
+    private static readonly TaskFactory TaskFactory = new(
+        CancellationToken.None,
+        TaskCreationOptions.LongRunning | TaskCreationOptions.PreferFairness,
```

Comment:
I would be careful with `LongRunning` here as we "jump" in front of the queue of the TP.
That also means that, potentially, we steal threads for HTTP requests.

If a user offloads tasks with **NcronJob** I guess milliseconds aren't important.

Reply:
Actually, using `LongRunning` doesn't cause tasks to "jump" in front of the queue. The `LongRunning` option indicates to the `TaskScheduler` that the task is expected to run for a long time, potentially prompting it to allocate additional threads to accommodate this. This doesn't prioritize the task over others but helps manage resources more effectively for long-running operations.

If you're referring to the priority queue, it's essentially unordered until the moment of dequeuing or peeking, at which point the job task that meets the JobQueueTupleComparer requirements is processed. This ensures that tasks are executed in the intended order based on their priority without stealing threads meant for other operations like HTTP requests.

Reply:
No, I really was referring to the TaskScheduler of .NET itself and how he handles those "virtual" threads.

Given your expertise, let's roll with it. In my lifetime my need for `LongRunning` was 0 until now.

Reply:
My concern is/was that we limit the scalability of the users application.
Or better: We prioritize "our" work (the jobs) over whatever is going on next to it.